js/util/script_detection.js

```diff
@@ -71,24 +102,181 @@ exports.charAllowsVerticalWritingMode = function(char, includeNeutrals) {
+            !(char >= 0xFFE8 /* halfwidth forms light vertical */ && char <= 0xFFEF)) {
+            return true;
+        }
+    }
+    if (isChar['Small Form Variants'](char)) {
+        if (!(char >= 0xFE58 /* small em dash */ && char <= 0xFE5E /* small right tortoise shell bracket */) &&
+            !(char >= 0xFE63 /* small hyphen-minus */ && char <= 0xFE66 /* small equals sign */)) {
+            return true;
+        }
+    }
+    if (isChar['Unified Canadian Aboriginal Syllabics'](char)) return true;
+    if (isChar['Unified Canadian Aboriginal Syllabics Extended'](char)) return true;
     if (isChar['Vertical Forms'](char)) return true;
+    if (isChar['Yijing Hexagram Symbols'](char)) return true;
+    if (isChar['Yi Syllables'](char)) return true;
+    if (isChar['Yi Radicals'](char)) return true;
 
-    if (isCharFullwidthForm(char)) return true;
+    // https://github.com/mapbox/mapbox-gl/issues/29
 
-    if (includeNeutrals) {
-        if (isChar['General Punctuation'](char)) return true;
-    }
+    //if (isChar['Meroitic Hieroglyphs'](char)) return true;
+    //if (isChar['Siddham'](char)) return true;
+    //if (isChar['Egyptian Hieroglyphs'](char)) return true;
+    //if (isChar['Anatolian Hieroglyphs'](char)) return true;
+    //if (isChar['Ideographic Symbols and Punctuation'](char)) return true;
+    //if (isChar['Tangut'](char)) return true;
+    //if (isChar['Tangut Components'](char)) return true;
+    //if (isChar['Kana Supplement'](char)) return true;
+    //if (isChar['Byzantine Musical Symbols'](char)) return true;
+    //if (isChar['Musical Symbols'](char)) return true;
+    //if (isChar['Tai Xuan Jing Symbols'](char)) return true;
+    //if (isChar['Counting Rod Numerals'](char)) return true;
+    //if (isChar['Sutton SignWriting'](char)) return true;
+    //if (isChar['Mahjong Tiles'](char)) return true;
+    //if (isChar['Domino Tiles'](char)) return true;
+    //if (isChar['Playing Cards'](char)) return true;
+    //if (isChar['Enclosed Alphanumeric Supplement'](char)) return true;
+    //if (isChar['Enclosed Ideographic Supplement'](char)) return true;
+    //if (isChar['Miscellaneous Symbols and Pictographs'](char)) return true;
+    //if (isChar['Emoticons'](char)) return true;
+    //if (isChar['Ornamental Dingbats'](char)) return true;
+    //if (isChar['Transport and Map Symbols'](char)) return true;
+    //if (isChar['Alchemical Symbols'](char)) return true;
+    //if (isChar['Geometric Shapes Extended'](char)) return true;
+    //if (isChar['Supplemental Symbols and Pictographs'](char)) return true;
+    //if (isChar['CJK Unified Ideographs Extension B'](char)) return true;
+    //if (isChar['CJK Unified Ideographs Extension C'](char)) return true;
+    //if (isChar['CJK Unified Ideographs Extension D'](char)) return true;
+    //if (isChar['CJK Unified Ideographs Extension E'](char)) return true;
+    //if (isChar['CJK Compatibility Ideographs Supplement'](char)) return true;
```

Comment:
I'd prefer if this code block were copied into https://github.com/mapbox/mapbox-gl/issues/29 than left commented out in the codebase.